test/framework/namespace.go

```diff
@@ -21,15 +21,21 @@ import (
 	"testing"
 
 	"github.com/pkg/errors"
+	"github.com/prometheus-operator/prometheus-operator/pkg/k8sutil"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 )
 
 func (f *Framework) CreateNamespace(ctx context.Context, t *testing.T, testCtx *TestCtx) string {
-	name := testCtx.GetObjID()
+	name := testCtx.ID()
+	rn := k8sutil.ResourceNamer{}
+	name, err := rn.UniqueVolumeName(name)
```

Comment:
maybe rename `UniqueVolumeName()` to `DNS1123Label()`?

Reply:
UniqueVolumeName was ok-ish because it was obvious that you would use it for volume names. UniqueName doesn't convey the fact that it returns a string which is a valid DNS label. The Kubernetes [documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names) already mentions `RFC 1123 Label Names`   and it's not even detailed which resources types follow this constraint :shrug:

Reply:
Alright, then I agree that it makes sense to go for `DNS1123Label` 👍

Reply:
In trying to make the functions clear, I've renamed:
- `UniqueVolumeName` -> `UniqueDNS1123Label`
- `VolumeName` -> `DNS1123Label`